integration/spark/app/src/test/java/io/openlineage/spark/agent/ConfigurableIntegrationTest.java

```diff
@@ -0,0 +1,210 @@
+      // append diff to test output files
+      fileLogger.write(eventVerifier.getDiffDescription());
+    }
+
+    fileLogger.flush();
+
+    if (testDir.endsWith(FAILING_TEST)) {
+      // assure the test is failing
+      assertThat(match).isFalse();
+    } else {
+      assertThat(match).isTrue();
+    }
+  }
+
+  @SneakyThrows
+  private static Stream<Arguments> provideTestDirs() {
+    return Files.list(Paths.get(System.getProperty("test.dir")))
+        .filter(path -> new File(path.toString()).isDirectory())
+        .filter(
+            path -> {
+              try {
+                boolean hasJsonFile =
+                    Files.list(path)
+                        .filter(p -> p.toString().toLowerCase(Locale.ROOT).endsWith(".json"))
+                        .findAny()
+                        .isPresent();
+
+                if (!hasJsonFile) {
+                  log.warn(
+                      "Test directory {} does not contain any JSON file to verify events emitted",
+                      path);
+                }
+
+                boolean hasJobFile =
+                    Files.list(path)
+                        .map(p -> p.toString().toLowerCase(Locale.ROOT))
+                        .filter(f -> f.endsWith(".sql") || f.endsWith(".py"))
+                        .findAny()
+                        .isPresent();
+
+                if (!hasJobFile) {
+                  log.warn("Test directory {} does not contain any job file (.sql or .py)", path);
+                }
+
+                return hasJobFile && hasJsonFile;
+              } catch (IOException e) {
+                log.error("Couldn't list directory {}", path);
+              }
+              return false;
+            })
+        .map(Arguments::of);
+  }
+
+  @SneakyThrows
+  private GenericContainer<?> makePysparkContainer(Path runDir, Writer output) {
+    List<String> command =
+        new ArrayList<>(
+            Arrays.asList(
+                "./bin/spark-submit",
+                "--master",
+                "local",
+                "--conf",
+                "spark.openlineage.transport.type=file",
+                "--conf",
+                "spark.openlineage.transport.location=" + runDir.toString() + "/events.json",
+                "--conf",
+                "spark.jars.ivy=" + OPENLINEAGE_ROOT_DIR + "/integration/spark/cli/.ivy2",
+                "--conf",
+                "spark.extraListeners=" + OpenLineageSparkListener.class.getName()));
+
+    Optional<Path> olJar =
+        Files.list(Paths.get(OPENLINEAGE_ROOT_DIR + "/integration/spark/app/build/libs/shadow"))
+            .filter(f -> f.getFileName().toString().endsWith(".jar"))
+            .findFirst();
+
+    if (config.getPackages() != null) {
+      command.add("--packages");
+      command.add(String.join(",", config.getPackages()));
+    }
+
+    command.add("--jars");
+    command.add(olJar.get().toAbsolutePath().toString());
+    command.add(getPySparkFile(runDir).getPath());
```

Comment:
in theory you could have just one python file called something like `runner.py` which would have only two lines
```
import sys
eval(sys.argv[1])
```
and you could supply the the python code that way, this saves you creating temp files and later reading from them. It could also add layer of complexity with possible escaping some characters or supplying arguments in general so idk if it's better or not

Reply:
Might work but passing Python code via command line parameter sounds like too evil hack compared to just adding a file :)